debezium-core/src/main/java/io/debezium/pipeline/metrics/DefaultSnapshotChangeEventSourceMetrics.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.pipeline.metrics;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.debezium.annotation.ThreadSafe;
+import io.debezium.connector.base.ChangeEventQueueMetrics;
+import io.debezium.connector.common.CdcSourceTaskContext;
+import io.debezium.pipeline.source.spi.EventMetadataProvider;
+import io.debezium.relational.TableId;
+import io.debezium.schema.DataCollectionId;
+
+/**
+ * The default implementation of metrics related to the snapshot phase of a connector.
+ *
+ * @author Randall Hauch, Jiri Pechanec
+ */
+@ThreadSafe
+public class DefaultSnapshotChangeEventSourceMetrics extends PipelineMetrics
+        implements SnapshotChangeEventSourceMetrics, SnapshotChangeEventSourceMetricsMXBean {
+
+    private final AtomicBoolean snapshotRunning = new AtomicBoolean();
+    private final AtomicBoolean snapshotCompleted = new AtomicBoolean();
+    private final AtomicBoolean snapshotAborted = new AtomicBoolean();
+    private final AtomicLong startTime = new AtomicLong();
```

Comment:
Depending on how adventureous you feel: how about changing this into `AtomicReference<Instant>` instead? One of the long-term refactoring goals is to move to the proper semantic time/date types from Java 8, and only convert from/to ms where it's needed at the system boundary. But I don't mean to bring this onto you as part of this work which isn't strictly requiring it. Throwing out the idea and preference nevertheless :)

Reply:
I feel adventurous but busy, so probably some time later :)

Reply:
Ok, no worries. I had to try :)